morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -0,0 +1,397 @@
+  mpretty (TraceLabelCreds _ a) = mpretty a
+
+instance
+  ( Signable (BftDSIGN c) (MorphoStdHeader h c),
+    MorphoStateDefaultConstraints h c
+  ) =>
+  MPretty (TraceForgeEvent (MorphoBlock h c))
+  where
+  mpretty (TraceAdoptedBlock slotNo blk txs) =
+    "Adopted forged block for slot"
+      <+> pretty (unSlotNo slotNo)
+      <> ":"
+      <+> mpretty (blockHash blk)
+      <> "; TxIds:"
+      <+> viaShow (map txId txs)
+  mpretty (TraceBlockFromFuture currentSlot tip) =
+    "Forged block from future: current slot"
+      <+> pretty (unSlotNo currentSlot)
+      <> ", tip being"
+      <+> mpretty tip
+  mpretty (TraceSlotIsImmutable slotNo tipPoint tipBlkNo) =
+    "Forged for immutable slot" <> pretty (unSlotNo slotNo)
+      <> ", tip:"
+      <+> pretty (showPoint MaximalVerbosity tipPoint)
+      <> ", block no:"
+      <+> pretty (unBlockNo tipBlkNo)
+  mpretty (TraceDidntAdoptBlock slotNo _) =
+    "Didn't adopt forged block at slot" <+> pretty (unSlotNo slotNo)
+  mpretty (TraceForgedBlock slotNo _ blk _) =
+    "Forged block for slot" <+> pretty (unSlotNo slotNo) <+> mpretty blk
+  mpretty (TraceForgedInvalidBlock slotNo _ reason) =
+    "Forged invalid block for slot"
+      <+> pretty (unSlotNo slotNo)
+      <> ", reason:"
+      <+> mpretty reason
+  mpretty (TraceNodeIsLeader slotNo) =
+    "Leading slot" <+> pretty (unSlotNo slotNo)
+  mpretty (TraceNodeNotLeader slotNo) =
+    "Not leading slot" <+> pretty (unSlotNo slotNo)
+  mpretty (TraceNodeCannotForge slotNo reason) =
+    "We are the leader for slot"
+      <+> pretty (unSlotNo slotNo)
+      <> ", but we cannot lead because:"
+      <+> viaShow reason
+  mpretty (TraceNoLedgerState slotNo _blk) =
+    "No ledger state at slot" <+> pretty (unSlotNo slotNo)
+  mpretty (TraceNoLedgerView slotNo _) =
+    "No ledger view at slot" <+> pretty (unSlotNo slotNo)
+  mpretty (TraceStartLeadershipCheck slotNo) =
+    "Testing for leadership at slot" <+> pretty (unSlotNo slotNo)
+  mpretty (TraceBlockContext slotNo _blk point) =
+    "The block to connect to is at slot"
+      <+> pretty (unSlotNo slotNo)
+      <> ", with point"
+      <+> viaShow point
+  mpretty (TraceLedgerState slotNo point) =
+    "The ledger state at the block to connect to is at slot"
+      <+> pretty (unSlotNo slotNo)
+      <> ", with point"
+      <+> viaShow point
+  mpretty (TraceLedgerView slotNo) =
+    "Ledger view for the current slot number"
+      <+> pretty (unSlotNo slotNo)
+      <+> "obtained"
+  mpretty (TraceForgeStateUpdateError slotNo err) =
+    "Updating the forge state at slot"
+      <+> pretty (unSlotNo slotNo)
+      <+> "failed with"
+      <+> viaShow err
+
+instance
+  (Signable (BftDSIGN c) (MorphoStdHeader h c), MorphoStateDefaultConstraints h c) =>
+  MPretty (ChainDB.TraceEvent (MorphoBlock h c))
+  where
+  mpretty (ChainDB.TraceAddBlockEvent ev) = case ev of
+    ChainDB.IgnoreBlockOlderThanK pt ->
+      "Ignoring block older than K:" <+> pretty (condenseT pt)
```

Comment:
In general, all `pretty (condenseT foo)` should be shortened to `mpretty foo`. In this case the `instance Condense (HeaderHash blk) => MPretty (Point blk)` will be useful